pkg/geo/geo.go

```diff
@@ -276,6 +278,52 @@ func (g *Geometry) CartesianBoundingBox() *CartesianBoundingBox {
 	return &CartesianBoundingBox{BoundingBox: *g.spatialObject.BoundingBox}
 }
 
+// SpaceCurveIndex returns an uint64 index to use representing an index into a Space Curve.
```

Comment:
Add some comments here about what happens in error or edge cases (like return 0 or math.MaxUint64). Also document a bit of what a Space Curve means, like, can you compare them? Is there a collision risk such that some objects will return the same value even though they are not actually equal? How does this function relate to CompareSpatialObject?

Reply:
Done